### PR TITLE
feat: add auto-generated header to generated files

### LINKED ISSUE #25: Emit "// <auto-generated/>" at top of file
The generated file produces a lot of warnings from StyleCop and other analyzers. Fixing all these is unreasonable. But all generated files should include a particular header that turns these off:

```cs
// <auto-generated/>
```

This one line at the very top of the file addresses almost everything. 
The only exception to this are the CS1591 and RS0016 warnings that #23 should address.

### LINKED ISSUE #55: Formatting generated c# files
While reading through the source code I noticed the following comment:

https://github.com/NordSecurity/uniffi-bindgen-cs/blob/5a4512f2d65ae7052533ac1c7a9733212a74d362/bindgen/src/lib.rs#L91-L92

I highly recommend taking a look at the tool [Csharpier](https://csharpier.com/), it is a lot faster and more reliable than `dotnet-format` and can format single files:

```bash
dotnet csharpier foo.cs
```

Note that files containg a `<auto-generated/>` header #25 will not be formatted, so the formatting stage must occur before writing the header.

### PR DESCRIPTION
**Pull request by meenzen**
closes #25, closes #55

### CODE REVIEW

File: bindgen/src/lib.rs [hunk 1]
```diff
@@ -86,10 +86,21 @@ impl uniffi_bindgen::BindingGenerator for BindingGenerator {
     ) -> anyhow::Result<()> {
         let bindings_file = out_dir.join(format!("{}.cs", ci.namespace()));
         let mut f = File::create(&bindings_file)?;
+
+        let version = env!("CARGO_PKG_VERSION");
+        let header = format!(
+            r#"// <auto-generated>
```
Comment (arg0d) [inline bindgen/src/lib.rs:92]: It feels like this should be part of [Wrapper.cs](https://github.com/NordSecurity/uniffi-bindgen-cs/blob/main/bindgen/templates/wrapper.cs). `CARGO_PKG_VERSION` can be added as `version` to `CsWrapper` struct, and accessed from the template using `{{ version }}`.

**Review by arg0d — CHANGES_REQUESTED** [review]
I see warnings in generated test code after adding auto generated annotation.
> _The annotation for nullable reference types should only be used in code within a '#nullable' annotations context. Auto-generated code requires an explicit '#nullable' directive in source._

I guess it should also be fixed?

**Comment by meenzen** [comment]
Having the header in wrapper.cs may make it harder to implement code formatting in the future, that's why I added this comment:

```cs
// Formatting must be done before the auto-generated header is added, otherwise the formatting tool will ignore the file.
```

**Comment by arg0d** [comment]
I see your point, but until formatting is implemented this looks very weird. I'm not familliar with C# formatting tools, but I would hope its possible to format an auto generated file.

**Comment by arg0d** [comment]
Oops, I somehow missed https://github.com/NordSecurity/uniffi-bindgen-cs/issues/55. Will take a quick look at the tool, maybe its psosible to format generated file.

**Comment by arg0d** [comment]
Looking at CSharpier code, its not possible to bypass the `auto-generated` guard. While you are at this, could you just implement the formatting using CSharpier? Its quite trivial, and it would clear up the reason why `auto-generated` is not in `wrapper.cs`.

[Formatting example from Kotlin](https://github.com/mozilla/uniffi-rs/blob/9f7c3c4663f024adff9c7b60fc3572a36a6eb136/uniffi_bindgen/src/bindings/kotlin/mod.rs#L28C10-L33C10).

**Comment by arg0d** [comment]
Formatting should be CLI option as well like in `mozilla/uniffi-rs` generators, in case the user does not have csharpier installed, or simply does not wish to format the file.

**Comment by meenzen** [comment]
Formatting is now implemented, but I'm not sure how to add the cli option. This is my first ever rust pr so I have no idea what I'm doing. Could you point me in the right direction?

**Comment by meenzen** [comment]
Sould I squash the commits and remove the changes to `Cargo.lock` and the `.gitignore` that snuck in?

**Pull request merged**